libtcmu.c

```diff
@@ -556,6 +558,7 @@ static int add_device(struct tcmulib_context *ctx, char *dev_name,
 	if (reopen && reset_supp)
 		tcmu_cfgfs_dev_exec_action(dev, "block_dev", 0);
 err_free:
+	if (dev->mmap_name) free(dev->mmap_name);
```

Comment:
See above comment. Fix up the gotos, so it goes to a specific unwinding and not a bulk free.